src/SymfonyCache/TaggableStore.php

```diff
@@ -0,0 +1,494 @@
+            foreach ($this->locks as $lock) {
+                $lock->release();
+            }
+        } catch (LockReleasingException $e) {
+            // noop
+        } finally {
+            $this->locks = [];
+        }
+    }
+
+    /**
+     * Remove/Expire cache objects based on cache tags.
+     * Returns true on success and false otherwise.
+     *
+     * @param array $tags Tags that should be removed/expired from the cache
+     *
+     * @return bool
+     */
+    public function invalidateTags(array $tags)
+    {
+        try {
+            return $this->cache->invalidateTags($tags);
+        } catch (InvalidArgumentException $e) {
+            return false;
+        }
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return string
+     */
+    public function getCacheKey(Request $request)
+    {
+        // Strip scheme to treat https and http the same
+        $uri = $request->getUri();
+        $uri = substr($uri, strlen($request->getScheme().'://'));
+
+        return 'md'.hash('sha256', $uri);
+    }
+
+    /**
+     * @param array     $vary
+     * @param HeaderBag $headerBag
+     *
+     * @return string
+     */
+    public function getVaryKey(array $vary, HeaderBag $headerBag)
+    {
+        if (0 === count($vary)) {
+            return self::NON_VARYING_KEY;
+        }
+
+        sort($vary);
+
+        $hashData = '';
+
+        foreach ($vary as $headerName) {
+            $hashData .= $headerName.':'.$headerBag->get($headerName);
+        }
+
+        return hash('sha256', $hashData);
+    }
+
+    /**
+     * @param Response $response
+     *
+     * @return string
+     */
+    public function generateContentDigest(Response $response)
+    {
+        return 'en'.hash('sha256', $response->getContent());
+    }
+
+    /**
+     * @param string $key
+     * @param string $data
+     * @param int    $expiresAfter
+     * @param array  $tags
+     *
+     * @return bool
+     */
+    private function saveDeferred($key, $data, $expiresAfter = null, $tags = [])
+    {
+        $item = $this->cache->getItem($key);
+        $item->set($data);
+        $item->expiresAfter($expiresAfter);
+
+        if (0 !== count($tags)) {
+            $item->tag($tags);
+        }
+
+        return $this->cache->saveDeferred($item);
+    }
+
+    /**
+     * Restores a Response from the cached data.
+     *
+     * @param array $cacheData An array containing the cache data
+     *
+     * @return Response|null
+     */
+    private function restoreResponse(array $cacheData)
+    {
+        $body = null;
+
+        if (isset($cacheData['headers']['x-content-digest'][0])) {
+            $item = $this->cache->getItem($cacheData['headers']['x-content-digest'][0]);
+            if ($item->isHit()) {
+                $body = $item->get();
+            }
+        }
+
+        return new Response(
+            $body,
+            $cacheData['status'],
+            $cacheData['headers']
+        );
+    }
+
+    /**
```

Comment:
can you add some text here? actually, i would like to change this method to getDefaultLockFactory and let it return a lock factory instead. with a phpdoc:

```
Build and return a default lock factory for when no explicit factory is specified.

The default factory uses the best quality lock store that is available on this system.
```